third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -822,14 +820,29 @@ impl<'env> Generator<'env> {
         }
     }
 
-    fn gen_arg_list(&mut self, exps: &[Exp]) -> Vec<TempIndex> {
-        exps.iter().map(|exp| self.gen_arg(exp)).collect()
+    /// Generate the code for a list of arguments.
+    fn gen_arg_list(&mut self, exps: &[Exp], force_temp: bool) -> Vec<TempIndex> {
```

Comment:
added `force_temp` to be passed to gen_arg to decide whether to enforce temp generation or not